### PR TITLE
VMware: Add CreateTemporaryDirectoryInGuest capability to vmware_guest_file_operation

### PR DESCRIPTION
**Pull request by jeking3**
##### SUMMARY

The current module allows the consumer to create and delete directories inside the guest.  The creation operation requires knowledge of the guest internal layout to succeed, since a full path is required.  The vSphere Guest Operations API also has a method called [`CreateTemporaryDirectoryInGuest`](https://code.vmware.com/apis/358/vsphere#/doc/vim.vm.guest.FileManager.html#createTemporaryDirectory) which eliminates that requirement.  This allows for easier interaction with the guest.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME

vmware_guest_file_operation

##### ANSIBLE VERSION

```paste below
ansible 2.8.0.dev0 (guest-operations 9b97f8bab3) last updated 2018/10/24 18:30:07 (GMT +000)
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /ansible/lib/ansible
  executable location = /ansible/bin/ansible
  python version = 2.7.15 (default, Sep 12 2018, 02:38:23) [GCC 6.4.0]
```

##### ADDITIONAL INFORMATION

- Added directory.operation `mktemp`
- Modified the options rules checking
- Added the temporary directory name to the result.

Example result:

```paste below
{
  "changed": true,
  "uuid": "42348510-53d2-d870-d6b0-61f1173123d2",
  "dir": "C:\\Users\\ADMINI~1\\AppData\\Local\\Temp\\ansiblevmware166test"
}
```

### CODE REVIEW

**Comment by jeking3** [comment]
Looks like this broke file upload:
```
Traceback (most recent call last):
  File "/ansible/lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py", line 443, in <module>
    main()
  File "/ansible/lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py", line 430, in main
    if module.params['directory']['operation'] in ('create', 'delete') and not module.params['directory']['path']:
TypeError: 'NoneType' object has no attribute '__getitem__'
```

File: lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py [hunk 1]
```diff
@@ -78,10 +78,12 @@
         required: True
     directory:
         description:
-            - Create or delete directory.
+            - Create or delete a directory.
             - 'Valid attributes are:'
-            - '  path: directory path to create or remove'
-            - '  operation: Valid values are create, delete'
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py:84]: Add notes like
```
- Also, can be used to create temp directory inside guest using mktemp operation.
- mktemp operation option is added in version 2.8
```

File: lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py [hunk 2]
```diff
@@ -218,16 +220,23 @@ def directory(self):
...
-        operation = self.module.params["directory"]['operation']
-        path = self.module.params["directory"]['path']
+        operation = self.module.params['directory']['operation']
+        path = self.module.params['directory']['path']
+        prefix = self.module.params['directory']['prefix']
+        suffix = self.module.params['directory']['suffix']
         creds = vim.vm.guest.NamePasswordAuthentication(username=vm_username, password=vm_password)
         file_manager = self.content.guestOperationsManager.fileManager
-        if operation == "create":
+        if operation == "create" or operation == "mktemp":
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py]:
```suggestion
        if operation in ("create", "mktemp"):
```

**Review by Akasurde — CHANGES_REQUESTED** [review]
Thanks for the contribution.

**Comment by Akasurde** [comment]
@pdellaert @dericcrago @Im0 @tchernomax @ckotte Could you please review this ? Thanks in advance.

**Review by Akasurde — APPROVED** [review]

**Comment by Akasurde** [comment]
LGTM, need another pair of eyes to take a look.

**Comment by jeking3** [comment]
Who needs to do this second review?

**Comment by Akasurde** [comment]
@pdellaert @dericcrago @Im0 @tchernomax @ckotte Could you please review this ? Thanks

**Comment by ckotte** [comment]
LGTM

**Comment by Akasurde** [comment]
@jeking3 Thanks for the contribution. @ckotte Thanks for the review.

**Pull request merged**